Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -1008,196 +1030,237 @@ namespace OpenInfraPlatform
-				//	ENTITY IfcRevolvedAreaSolid
-				//		SUPERTYPE OF(IfcRevolvedAreaSolidTapered)
-				//		SUBTYPE OF(IfcSweptAreaSolid);
-				//			Axis: IfcAxis1Placement;
-				//			Angle: IfcPlaneAngleMeasure;
-				//		DERIVE
-				//			AxisLine : IfcLine: = IfcRepresentationItem() || IfcGeometricRepresentationItem() || IfcCurve() || IfcLine(Axis.Location, IfcRepresentationItem() || IfcGeometricRepresentationItem() || IfcVector(Axis.Z, 1.0));
-				//		WHERE
-				//			AxisStartInXY : Axis.Location.Coordinates[3] = 0.0;
-				//			AxisDirectionInXY: Axis.Z.DirectionRatios[3] = 0.0;
-				//	END_ENTITY;
-				// **************************************************************************************************************************
-
-				if (revolvedArea.expired())
-					throw oip::ReferenceExpiredException(revolvedArea);
-
+#ifdef _DEBUG
+				BLUE_LOG(trace) << "Processing IfcRevolvedAreaSolid #" << revolvedArea->getId();
+#endif
 				// Get axis and angle (attributes 3-4). 
-				oip::EXPRESSReference<typename IfcEntityTypesT::IfcAxis1Placement> axis = revolvedArea->Axis;
+				std::shared_ptr<typename IfcEntityTypesT::IfcAxis1Placement> axis =
+					revolvedArea->Axis.lock();
+				
+
+				if (!revolvedArea->SweptArea)
+				{
+					return;
+				}
+				double length_factor = UnitConvert()->getLengthInMeterFactor();
 
 				// angle and axis
-				oip::EXPRESSReference<typename IfcEntityTypesT::IfcProfileDef> sweptAreaProfile = revolvedArea->SweptArea;
-				double revolutionAngle = revolvedArea->Angle * this->UnitConvert()->getAngleInRadianFactor();
+				double angle_factor = UnitConvert()->getAngleInRadianFactor();
+				oip::EXPRESSReference<typename IfcEntityTypesT::IfcProfileDef> sweptArea_profile = revolvedArea->SweptArea;
+				double revolution_angle = revolvedArea->Angle * angle_factor;
 
-				carve::geom::vector<3>  axisLocation;
-				carve::geom::vector<3>  axisDirection;
+				carve::geom::vector<3>  axis_location;
+				carve::geom::vector<3>  axis_direction;
 				if (revolvedArea->Axis)
 				{
-					oip::EXPRESSReference<typename IfcEntityTypesT::IfcAxis1Placement> axisPlacement = revolvedArea->Axis;
+					std::shared_ptr<typename IfcEntityTypesT::IfcAxis1Placement> axis_placement = revolvedArea->Axis.lock();
 
-					if (axisPlacement->Location)
-						axisLocation = placementConverter->convertIfcPoint(axisPlacement->Location);
-					
-					if (axisPlacement->Axis)
-						axisDirection = placementConverter->convertIfcDirection(axisPlacement->Axis);
+					if (axis_placement->Location)
+					{
+						axis_location = placementConverter->convertIfcCartesianPoint(axis_placement->Location );
+					}
+
+					if (axis_placement->Axis)
+					{
+						decltype(axis_placement->Axis)::type axis = axis_placement->Axis;
+						axis_direction = carve::geom::VECTOR(
+							axis->DirectionRatios[0],
+							axis->DirectionRatios[1],
+							axis->DirectionRatios[2]);
+					}

```

Comment:
This was fine already - I assume that it got reverted?

Reply:
Resolved: 57e281f